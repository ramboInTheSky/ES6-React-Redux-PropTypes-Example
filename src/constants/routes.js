import devElse from '../util/devElse';
import {
  activitiesApi as activitiesApiBase,
  arrearsApi as arrearsApiBase,
  avatarsApi as avatarsApiBase,
  correspondenceApi as correspondenceApiBase,
  customerDashboard as customerDashboardBase,
  customersApi as customersApiBase,
  documentsApi as documentsApiBase,
  employeeDashboard as employeeDashboardBase,
  interactionsApi as interactionsApiBase,
  notesApi as notesApiBase,
  paymentsApi as paymentsApiBase,
  paymentsBaseURL as paymentsBase,
  tasksApi as tasksApiBase,
  sharepointApi as sharepointApiBase,
} from '../constants/tokens';

export const activitiesApi = devElse(
  'https://mynottinghill-dev.workwise.london/activities/web/api/v1',
  activitiesApiBase
);

export const arrearsApi = devElse(
  'https://mynottinghill-dev.workwise.london/arrears/web/api/v1',
  arrearsApiBase
);

export const avatarsApi = devElse(
  'https://nhhuksmedsamediad.blob.core.windows.net/employees/$id/Image/profile.jpg/profile_original.jpg',
  avatarsApiBase
);

export const correspondenceApi = devElse(
  'https://mynottinghill-dev.workwise.london/correspondence/web/api/v1/',
  correspondenceApiBase
);

export const customersApi = devElse(
  'https://mynottinghill-dev.workwise.london/customers/web/api/v1',
  customersApiBase
);

export const employeeDashboard = devElse(
  'https://mynottinghill-dev.workwise.london/dashboards/employees',
  employeeDashboardBase
);

export const customerDashboard = devElse(
  'https://mynottinghill-dev.workwise.london/dashboards/customers',
  customerDashboardBase
);

export const interactionsApi = devElse(
  'https://mynottinghill-dev.workwise.london/interactions/web/api/v1',
  interactionsApiBase
);

export const notesApi = devElse(
  'https://mynottinghill-dev.workwise.london/notes/web/api/v1',
  notesApiBase
);

export const payments = devElse(
  'https://mynottinghill-dev.workwise.london/payments/web',
  paymentsBase
);

export const paymentsApi = devElse(
  'https://mynottinghill-dev.workwise.london/payments/web/api/v1',
  paymentsApiBase
);

export const tasksApi = devElse(
  'https://mynottinghill-dev.workwise.london/tasks/web/api/v1',
  tasksApiBase
);

export const documentsApi = devElse(
  'https://mynottinghill-dev.workwise.london/documents/web/api/v1',
  documentsApiBase
);

export const sharepointApi = devElse(
  'https://mynottinghill-dev.workwise.london/documents/sharepoint',
  sharepointApiBase
);
