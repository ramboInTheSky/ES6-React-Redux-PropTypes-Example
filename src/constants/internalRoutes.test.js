import routes, { getTaskActivityRouteByActionType } from './internalRoutes';

describe('internalRoutes', () => {
  it('generates the correct routes when a payment plan exists', () => {
    const arrearDetails = {
      id: 'abc123',
      paymentPlan: { id: 'def456' },
    };
    expect(routes(arrearDetails)).toMatchSnapshot();
  });

  it('generates the correct routes when there is an active pause', () => {
    const arrearDetails = {
      id: 'abc123',
      pauseSummary: [{ id: 'def456', status: 'Active' }],
    };
    expect(routes(arrearDetails)).toMatchSnapshot();
  });

  it('generates the correct routes when there is a pause awaiting approval and user is an HO', () => {
    const arrearDetails = {
      id: 'abc123',
      pauseSummary: [{ id: 'def456', status: 'Draft' }],
    };
    expect(routes(arrearDetails)).toMatchSnapshot();
  });

  it('generates the correct routes when there is a pause awaiting approval and user is an HM', () => {
    const arrearDetails = {
      id: 'abc123',
      pauseSummary: [{ id: 'def456', status: 'Draft' }],
    };
    expect(routes(arrearDetails, true)).toMatchSnapshot();
  });

  it('generates the correct routes for add interaction', () => {
    expect(getTaskActivityRouteByActionType('addinteraction', 'abc123')).toEqual(
      '/arrears-details/abc123/interaction/create'
    );
  });

  it('generates the correct routes for add note', () => {
    expect(getTaskActivityRouteByActionType('addnote', 'abc123')).toEqual(
      '/arrears-details/abc123/note/create'
    );
  });

  it('generates the correct routes for send correspondence', () => {
    expect(getTaskActivityRouteByActionType('sendcorrespondence', 'abc123')).toEqual(
      '/arrears-details/abc123/send-correspondence/create'
    );
  });

  it('generates the correct routes for legal referral when legalReferralId is NOT provided', () => {
    expect(getTaskActivityRouteByActionType('legalreferral', 'abc123')).toEqual(
      '/arrears-details/abc123/legal-case-referral/create'
    );
  });

  it('generates the correct routes for legal referral when legalReferralId IS provided', () => {
    expect(
      getTaskActivityRouteByActionType('legalreferral', 'ARREARSID', 'LEGALREFERRALID')
    ).toEqual('/arrears-details/ARREARSID/legal-case-referral/LEGALREFERRALID/review');
  });
});
