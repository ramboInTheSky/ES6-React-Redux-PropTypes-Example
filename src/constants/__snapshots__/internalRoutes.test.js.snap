// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`internalRoutes generates the correct routes when a payment plan exists 1`] = `
Object {
  "addNote": "/arrears-details/abc123/note/create",
  "addTask": "/arrears-details/abc123/task/create",
  "arrearsDetailNOSPModal": "/arrears-details/abc123?modal=serveNOSP",
  "interaction": "/arrears-details/abc123/interaction/create",
  "legalReferral": "/arrears-details/abc123/legal-case-referral/create",
  "pause": "/arrears-details/abc123/pause-arrears-case/create",
  "paymentPlan": "/arrears-details/abc123/payment-plan/def456",
  "sendCorrespondence": "/arrears-details/abc123/send-correspondence/create",
  "tenancySustainmentReferral": "/arrears-details/abc123/refer-arrears-case/create",
}
`;

exports[`internalRoutes generates the correct routes when there is a pause awaiting approval and user is an HM 1`] = `
Object {
  "addNote": "/arrears-details/abc123/note/create",
  "addTask": "/arrears-details/abc123/task/create",
  "arrearsDetailNOSPModal": "/arrears-details/abc123?modal=serveNOSP",
  "interaction": "/arrears-details/abc123/interaction/create",
  "legalReferral": "/arrears-details/abc123/legal-case-referral/create",
  "pause": "/arrears-details/abc123/pause-arrears-case/def456/approve",
  "paymentPlan": "/arrears-details/abc123/payment-plan/create",
  "sendCorrespondence": "/arrears-details/abc123/send-correspondence/create",
  "tenancySustainmentReferral": "/arrears-details/abc123/refer-arrears-case/create",
}
`;

exports[`internalRoutes generates the correct routes when there is a pause awaiting approval and user is an HO 1`] = `
Object {
  "addNote": "/arrears-details/abc123/note/create",
  "addTask": "/arrears-details/abc123/task/create",
  "arrearsDetailNOSPModal": "/arrears-details/abc123?modal=serveNOSP",
  "interaction": "/arrears-details/abc123/interaction/create",
  "legalReferral": "/arrears-details/abc123/legal-case-referral/create",
  "pause": "/arrears-details/abc123/pause-arrears-case/def456/update",
  "paymentPlan": "/arrears-details/abc123/payment-plan/create",
  "sendCorrespondence": "/arrears-details/abc123/send-correspondence/create",
  "tenancySustainmentReferral": "/arrears-details/abc123/refer-arrears-case/create",
}
`;

exports[`internalRoutes generates the correct routes when there is an active pause 1`] = `
Object {
  "addNote": "/arrears-details/abc123/note/create",
  "addTask": "/arrears-details/abc123/task/create",
  "arrearsDetailNOSPModal": "/arrears-details/abc123?modal=serveNOSP",
  "interaction": "/arrears-details/abc123/interaction/create",
  "legalReferral": "/arrears-details/abc123/legal-case-referral/create",
  "pause": "/arrears-details/abc123/pause-arrears-case/def456/update",
  "paymentPlan": "/arrears-details/abc123/payment-plan/create",
  "sendCorrespondence": "/arrears-details/abc123/send-correspondence/create",
  "tenancySustainmentReferral": "/arrears-details/abc123/refer-arrears-case/create",
}
`;
